src/test/java/core/basesyntax/ShopServiceTest.java

```diff
@@ -0,0 +1,104 @@
+package core.basesyntax;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.services.ShopService;
+import core.basesyntax.services.impl.ShopServiceImpl;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import core.basesyntax.strategy.operations.BalanceOperation;
+import core.basesyntax.strategy.operations.OperationHandler;
+import core.basesyntax.strategy.operations.PurchaseOperation;
+import core.basesyntax.strategy.operations.ReturnOperation;
+import core.basesyntax.strategy.operations.SupplyOperation;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ShopServiceTest {
+    private static ShopService shopService;
+    private static List<FruitTransaction> soloListFruit;
+    private static List<FruitTransaction> multiplyListFruits;
+
+    @BeforeAll
+    public static void setUp() {
+        soloListFruit = new ArrayList<>();
+        multiplyListFruits = new ArrayList<>();
+
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlers = new HashMap<>();
+        operationHandlers.put(FruitTransaction.Operation.BALANCE, new BalanceOperation());
+        operationHandlers.put(FruitTransaction.Operation.PURCHASE, new PurchaseOperation());
+        operationHandlers.put(FruitTransaction.Operation.RETURN, new ReturnOperation());
+        operationHandlers.put(FruitTransaction.Operation.SUPPLY, new SupplyOperation());
+        OperationStrategy operationStrategy = new OperationStrategyImpl(operationHandlers);
+        shopService = new ShopServiceImpl(operationStrategy);
+    }
+
+    @BeforeEach
```

Comment:
clearing should be in AfterEach
```suggestion
    @AfterEach
```